cmd/security-secrets-setup/entrypoint.sh

```diff
@@ -24,53 +24,36 @@ set -e
 
 # runtime directory is set per user:
-export XDG_RUNTIME_DIR=${XDG_RUNTIME_DIR:-/run/user/$(echo $(id -u))}
-
-PKI_INIT_RUNTIME_DIR=${XDG_RUNTIME_DIR}/${PKI_INIT_DIR}
+XDG_RUNTIME_DIR=${XDG_RUNTIME_DIR:-/run/user/$(echo $(id -u))}
+PATH="$BASE_DIR:$PATH"
+VAULT_TLS_PATH=${VAULT_TLS_PATH:-/run/edgex/secrets/edgex-vault}
+export XDG_RUNTIME_DIR PATH VAULT_TLS_PATH
 
 # debug output:
 echo XDG_RUNTIME_DIR $XDG_RUNTIME_DIR
-echo PKI_INIT_RUNTIME_DIR $PKI_INIT_RUNTIME_DIR
-
-# configuration for TLS materials:
-PKI_CONFIG_JSON_DIR="res"
-PKI_SETUP_VAULT_FILE=${PKI_CONFIG_JSON_DIR}"/pkisetup-vault.json"
-PKI_SETUP_KONG_FILE=${PKI_CONFIG_JSON_DIR}"/pkisetup-kong.json"
+echo BASE_DIR $BASE_DIR
 
-# check if files exists
-if [ ! -f "${PKI_SETUP_VAULT_FILE}" ]; then
-    echo "Error: certificate config file for Vault is missing"
-    exit 1
+# if running security-secrets-setup subcommand
+# build full command line into positional args
+if [ "$1" = 'generate' -o "$1" = 'cache' -o "$1" = 'import' -o "$1" = 'legacy' ]; then
+    set -- security-secrets-setup "$@"
 fi
 
-if [ ! -f "${PKI_SETUP_KONG_FILE}" ]; then
-    echo "Error: certificate config file for Kong is missing"
-    exit 1
+instvaultscript=""
+posthook=""
+if [ "$1" = 'security-secrets-setup' ]; then
+    # update the start_vault script for vault starting
+    instvaultscript="mv /vault/staging/start_vault.sh /vault/init/start_vault.sh"
+    # grant permissions of folders for vault:vault
+    posthook="chown -R 100:1000 ${VAULT_TLS_PATH}/ /vault/init/"
 fi
 
-# the working dir should be in vault dir based upon the current Docker image
-BASE_DIR="${BASE_DIR:-/vault}"
-cd $BASE_DIR
-CERT_DIR=$(jq -r '.working_dir' ${PKI_SETUP_VAULT_FILE})
-CERT_SUBDIR=$(jq -r '.pki_setup_dir' ${PKI_SETUP_VAULT_FILE})
-ROOT_NAME=$(jq -r '.x509_root_ca_parameters | .ca_name' ${PKI_SETUP_VAULT_FILE})
-CERT_EXEC="${CERT_EXEC:-./security-secrets-setup}"
-# check to see if the root certificate generate with security-secrets-setup already exists
-# if so then do not generate a new set of them
-if [ ! -f "$CERT_DIR/$CERT_SUBDIR/$ROOT_NAME/$ROOT_NAME.pem" ]; then
-    ${CERT_EXEC} legacy --config ${PKI_SETUP_VAULT_FILE}
-    [ $? -eq 0 ] || (echo "failed to generate TLS assets for Vault" && exit 1)
-
-    ${CERT_EXEC} legacy --config ${PKI_SETUP_KONG_FILE}
-    [ $? -eq 0 ] || (echo "failed to generate TLS assets for Kong" && exit 1)
+echo "Executing $@"
+"$@"
 
-    # delete CA private key
-    rm "$PWD/$CERT_DIR/$CERT_SUBDIR/$ROOT_NAME/$ROOT_NAME.priv.key"
-    [ $? -eq 0 ] || (echo "failed to delete sensitive CA private key" && exit 1)
-
-    # take ownership
-    chown -R vault:vault $PWD/$CERT_DIR/$CERT_SUBDIR || true
-fi 
-
-# run the Vault's docker-entry script 
-source /usr/local/bin/docker-entrypoint.sh
+if [ "$1" = 'security-secrets-setup' ]; then
+    echo "Installing Vault's startup script"
+    $instvaultscript
+    echo "Executing hook=$posthook"
+    $posthook
+fi
```

Comment:
Tehnically I think the `if` here is redundant, but you can leave it.

Reply:
not really, if I type `exit` command inside the container, it will execute this two commands which I really do not want to

Reply:
k